internal/schema/schema.go

```diff
@@ -348,41 +350,110 @@ func (t Trigger) GetName() string {
 
 // GetPublicSchema fetches the "public" schema. It is a non-atomic operation.
 func GetPublicSchema(ctx context.Context, db queries.DBTX) (Schema, error) {
-	q := queries.New(db)
+	// To allow backwards compatibility with connections, we will not use concurrency if passed in a db that is not a
+	// *sql.DB. This is because not all implementations are thread safe, e.g., pgx.Connection.
+	//
+	// In the future, we should maybe create options where users can pass in a DB pool (WithPool(db) or WithConnection(db))
+	// and we can set concurrency to 1 if the passed in db is not a *sql.DB.
+	goRoutineRunner := util.NewSynchronousGoRoutineRunner()
+	if _, ok := db.(*sql.DB); ok {
+		goRoutineRunner = util.NewGoroutineLimiter(50)
```

Comment:
Bound the number of go routines we start. This isn't strictly necessary, but is probably a good practice.